components/merkledb/src/backends/temporarydb.rs

```diff
@@ -30,15 +30,20 @@ use crate::{db::DB_METADATA, Database, DbOptions, Patch, Result, Snapshot};
 #[derive(Debug)]
 pub struct TemporaryDB {
     inner: RocksDB,
-    dir: TempDir,
+    dir: Arc<TempDir>,
+}
+
+struct TemporarySnapshot {
```

Comment:
Nit: I'd add a docstring with some description like "`RocksDB` snapshot wrapper with the `TempDir` handle to prevent it from removing until all the snapshots and database itself are dropped"